plugins/http_plugin/include/eosio/http_plugin/beast_http_session.hpp

```diff
@@ -346,19 +342,29 @@ class beast_http_session : public detail::abstract_conn {
          res_->set(http::field::content_type, "application/json");
          res_->keep_alive(false);
          res_->set(http::field::server, BOOST_BEAST_VERSION_STRING);
-
+         
          send_response(std::move(err_str), static_cast<unsigned int>(http::status::internal_server_error));
          derived().do_eof();
       }
    }
 
-   virtual void send_response(std::string json_body, unsigned int code) override {
+   void increment_bytes_in_flight(size_t sz) {
+      plugin_state_->bytes_in_flight += sz;
+   }
+
+   void decrement_bytes_in_flight(size_t sz) {
+      plugin_state_->bytes_in_flight -= sz;
```

Comment:
I guess `bytes_in_flight` would never be smaller than `sz` before subtract.

Reply:
That is correct, we always decrement after incrementing with the same value.